sonar-db/src/main/java/org/sonar/db/user/UserTokenDao.java

```diff
@@ -53,6 +58,22 @@ public UserTokenDto selectOrFailByTokenHash(DbSession dbSession, String tokenHas
     return mapper(dbSession).selectByLogin(login);
   }
 
+  public Map<String, Integer> countTokensByLogins(final DbSession dbSession, List<String> logins) {
+    final Map<String, Integer> result = new HashMap<>();
+    DatabaseUtils.executeLargeInputs(logins, new Function<List<String>, List<UserTokenCount>>() {
+      @Override
+      public List<UserTokenCount> apply(@Nonnull List<String> input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)